src/onemod/types/filepath.py

```diff
@@ -0,0 +1,14 @@
+from pydantic import BaseModel, FilePath as PydanticFilePath, model_validator
+
+
+class FilePath(BaseModel):
+    path: PydanticFilePath
+    extension: str | None = None
+    
+    @model_validator(mode="after")
+    def check_extension(self):
```

Comment:
`@model_validator` vs. `model_post_init`: Is the advantage that the function name makes it clear what it does? I guess when I've used `model_post_inti`, I was often doing things other than validation.